### PR TITLE
Split Console and Platform into new libraries

### PR DESCRIPTION
**Pull request by ZehMatt**
I just moved the Console code as is and I would prefer we move one day to something like spdlog or we roll our own multi-sink logging, either way works for me, but Console should not directly print to anything and instead print into sinks and they can print the output to whatever.

Also the CMake source/header files were not properly sorted since I added my file just at the end and re-sorted I didn't expect it to change that much, so it is what it is unfortunately.

### CODE REVIEW

File: src/Platform/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+set(private_files
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Crash.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Platform.Posix.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/Platform.Windows.cpp"
+)
+
+if (APPLE)
+    set(private_files 
+        "${CMAKE_CURRENT_SOURCE_DIR}src/Platform/Platform.Macos.mm")
```
Comment (duncanspumpkin) [inline src/Platform/CMakeLists.txt:14]: extra `Platform` in the address and you are missing a `/`

File: src/OpenLoco/CMakeLists.txt [hunk 2]
```diff
@@ -428,63 +434,45 @@ set(OLOCO_HEADERS
...
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/Ui/UiTypes.hpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/Viewport.hpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/ZoomLevel.hpp")
 
-if (APPLE)
-    set(OLOCO_MM_SOURCES 
-        "${CMAKE_CURRENT_SOURCE_DIR}src/Platform/Platform.Macos.mm")
-    set_source_files_properties(${OLOCO_MM_SOURCES} PROPERTIES COMPILE_FLAGS "-x objective-c++")
-endif()
-
```
Comment (duncanspumpkin) [inline src/OpenLoco/CMakeLists.txt:487]: L492 should remove the OLOCO_MM_SOURCES

File: src/Platform/CMakeLists.txt (hunk 1 above)
Comment (duncanspumpkin) [reply src/Platform/CMakeLists.txt:14]: Oh and also this should append to the private_files not set the private files.

Comment (ZehMatt) [reply src/Platform/CMakeLists.txt:14]: I took this as is from the OpenLoco CMakeLists didn't give it much thought, I'll look into it.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**